shared/src/commonMain/kotlin/org/hyperskill/app/gamification_toolbar/injection/GamificationToolbarComponentImpl.kt

```diff
@@ -0,0 +1,20 @@
+package org.hyperskill.app.gamification_toolbar.injection
+
+import org.hyperskill.app.core.injection.AppGraph
+import org.hyperskill.app.core.presentation.ActionDispatcherOptions
+import org.hyperskill.app.gamification_toolbar.presentation.GamificationToolbarActionDispatcher
+import org.hyperskill.app.gamification_toolbar.presentation.GamificationToolbarReducer
+
+class GamificationToolbarComponentImpl(private val appGraph: AppGraph) : GamificationToolbarComponent {
+    override val gamificationToolbarReducer: GamificationToolbarReducer
+        get() = GamificationToolbarReducer()
+
+    override val gamificationToolbarActionDispatcher: GamificationToolbarActionDispatcher
+        get() = GamificationToolbarActionDispatcher(
+            ActionDispatcherOptions(),
+            appGraph.buildProfileDataComponent().profileInteractor,
+            appGraph.buildStreaksDataComponent().streaksInteractor,
```

Comment:
The same thing as with `TrackComponentImpl`, let's buildd componente once.

Reply:
I suppose here you meant `HomeComponentImpl`, not `GamificationToolbarComponentImpl`